src/Tests/Parser/Dothtml/DothtmlParserTests.cs

```diff
@@ -260,21 +260,23 @@ public void DothtmlParser_Invalid_SpaceInTagName()
         {
             var markup = @"<dot:ContentPlace Holder DataContext=""sdads"">
 </ dot:ContentPlaceHolder > ";
-            var nodes = ParseMarkup(markup).Content;
+            var root = ParseMarkup(markup);
+            var nodes = root.Content;
 
-            Assert.AreEqual(3, nodes.Count);
+            Assert.IsTrue(root.EnumerateNodes().Any(n => n.HasNodeErrors || n.Tokens.Any(t => t.HasError)), "this is invalid code, there should be error on some node");
+
+            Assert.AreEqual(2, nodes.Count);
```

Comment:
I also slightly changed what happens in the case of incomplete close tag - it now closes the current element, regarless of its name. However, this cannot break any running code, since incomplete end tag has a token error and thus the compilation fails.